protocol/git/README.md

```diff
@@ -1,120 +1,50 @@
-web interface or in the project's chat room.
-
-For changes which they can make themselves, they check out the branch.
-
-    git checkout <branch-name>
-    ./bin/setup
-    git diff staging/master..HEAD
-
-They make small changes right in the branch, test the feature on their machine,
-run tests, commit, and push.
-
-When satisfied, they comment on the pull request `Ready to merge.`
-
-Merge
------
-
-Rebase interactively. Squash commits like "Fix whitespace" into one or a
-small number of valuable commit(s). Edit commit messages to reveal intent. Run
-tests.
-
-    git fetch origin
-    git rebase -i origin/master
-
-Force push your branch. This allows GitHub to automatically close your pull
-request and mark it as merged when your commit(s) are pushed to master. It also
- makes it possible to [find the pull request] that brought in your changes.
-
-    git push --force-with-lease origin <branch-name>
-
-View a list of new commits. View changed files. Merge branch into master.
-
-    git log origin/master..<branch-name>
-    git diff --stat origin/master
-    git checkout master
-    git merge <branch-name> --ff-only
-    git push
-
-Delete your remote feature branch.
-
-    git push origin --delete <branch-name>
-
-Delete your local feature branch.
-
-    git branch --delete <branch-name>
-
-[find the pull request]: http://stackoverflow.com/a/17819027
+# Proposal: Use release branches and cease automatic staging
+
+Our current process for integrating, delivering, and deploying to healthify:
+
+1. Create a changeset on a development branch forked from `master`.
+2. Submit a pull request to merge the branch into `master`.
+3. Review the changes in the pull request.
+4. Merge the development branch into `master`, triggering delivery to staging.
+5. Conduct manual QA through the front end of the staging app.
+
+This process is conducted _concurrently_ for multiple changesets. If we have at least one changeset that has passed all 5 steps, and none that are between steps 4 and 5, we are ready to deploy.
+
+We recognize three resulting problems:
+
+1. We cannot deploy when there is any changeset that has finished 1-4 but not 5.
+2. Sometimes, features are constituted by many distinct changesets.
+3. Sometimes, a set of many features should be deployed simultaneously.
+
+Therefore, we propose a two-part solution: Use release branches to integrate multiple dependent changesets, and decouple the post-code-review integration from the staging deployment.
+
+1. Use release branches. Every branch to be merged into master should represent an atomically deployable feature set (i.e,. a release). Changes should be committed on _branches off of the release branch_ as opposed to branches off of master. There should be at least one development branch for each release branch; there can be multiple development branches for a release branch, to map better to Pivotal stories or to yield more digestible PRs.
```

Comment:
> nit-pick: master instead of master to be consistent with the rest of the file
+1

> removing "to map better to Pivotal stories"

I think we can remove the phrase without any significant loss of communication value.

Just to clarify, though: We want release branches to map 1:1 with deployable feature-sets. Therefore, if we have several PT cards that can only be deployed as a set, the release branch would represent the complete set, and its development branches would each represent one of the constituent PT cards. Alternatively, we could be careful not to subdivide PT cards into features that aren't independently deployable.

I'm thinking of the caregiver epic in particular here: Would we really have a single PT card for creating a caregiver, associating an existing caregiver, and removing a caregiver association? None of those features could be reasonably deployed without all of them available, but having separate cards helped us organize who was working on what.